www/guides/react.md

```diff
@@ -0,0 +1,373 @@
+
+## Organizing your files
+
+If you're going to be adding a bunch of new files, you probably don't want them in the root directory. A standard way of organizing React projects is an `src` directory for components and their associated files. Then a public directory for static assets like images, fonts, icons, and sometimes CSS. To do this you'll need to reorganize your files, then tell Snowpack about it in the configuration file, and update your paths in index.html.
+
+```
+📁 src
+↳ index.jsx
+📁 public
+↳ index.css
+↳ index.html
+```
+
+```bash
+mkdir src
+mkdir public
+mv index.jsx src/index.jsx
+mv index.html public/index.html
+mv index.css public/index.css
+```
+
+You need to tell Snowpack about this new structure using the [Snowpack Configuration API](https://www.snowpack.dev/#configuration) in the `snowpack.config.js` file. Right now it's a bare bones scaffold that doesn't do anything. Add this to the `mount` object:
+
+```diff
+  mount: {
+-    /* ... */
++    public: '/',
++    src: '/_dist_',
+  },
+```
+
+This tells it look at `public` instead of look at `/` for the index. For example when building `index.css`, Snowpack looks for this files in `./public/index.css`.
+
+And it also tells it to put the final output files in a subdirectory called `_dist_` so change that path in `index.html` from `/index.js` to `/_dist_/index.js`.
+
+```diff
+    <h1>Welcome to Snowpack!</h1>
+    <div id="root"></div>
+-    <script type="module" src="/index.js"></>
++    <script type="module" src="/_dist_/index.js"></script>
+  </body>
+```
+
+If Snowpack is still running, exit the process, you need to restart Snowpack for configuration file changes. When you start up again, if it worked, it should look the same.
+
+Create a new file at `src/App.jsx` and paste the following code into this new file to create an `App` component:
+
+```jsx
+import React, {useState, useEffect} from 'react';
+
+function App() {
+  // Create the count state.
+  const [count, setCount] = useState(0);
+  // Create the counter (+1 every second).
+  useEffect(() => {
+    const timer = setTimeout(() => setCount(count + 1), 1000);
+    return () => clearTimeout(timer);
+  }, [count, setCount]);
+  // Return the App component.
+  return (
+    <div className="App">
+      <header className="App-header">
+        <p>
+          Edit <code>src/App.jsx</code> and save to reload.
+        </p>
+        <p>
+          Page has been open for <code>{count}</code> seconds.
+        </p>
+        <p>
+          <a
+            className="App-link"
+            href="https://reactjs.org"
+            target="_blank"
+            rel="noopener noreferrer"
+          >
+            Learn React
+          </a>
+        </p>
+      </header>
+    </div>
+  );
+}
+
+export default App;
+```
+
+Now include it in `index.js`
+
+```diff
+import React from 'react';
+import ReactDOM from 'react-dom';
++ import App from './App';
```

Comment:
isn't it `App.jsx` or does snowpack do magic here?

Reply:
I'm pretty sure Snowpack does magic. But @FredKSchott would know more.

Reply:
Both are supported, but its best to get users in the habit of including the file extension